ddtrace/llmobs/_llmobs.py

```diff
@@ -1040,99 +1070,115 @@ def submit_evaluation_for(
             )
             return
 
-        has_exactly_one_joining_key = (span is not None) ^ (span_with_tag_value is not None)
-
-        if not has_exactly_one_joining_key:
-            raise ValueError(
-                "Exactly one of `span` or `span_with_tag_value` must be specified to submit an evaluation metric."
-            )
-
+        error = None
         join_on = {}
-        if span is not None:
-            if (
-                not isinstance(span, dict)
-                or not isinstance(span.get("span_id"), str)
-                or not isinstance(span.get("trace_id"), str)
-            ):
-                raise TypeError(
-                    "`span` must be a dictionary containing both span_id and trace_id keys. "
-                    "LLMObs.export_span() can be used to generate this dictionary from a given span."
-                )
-            join_on["span"] = span
-        elif span_with_tag_value is not None:
-            if (
-                not isinstance(span_with_tag_value, dict)
-                or not isinstance(span_with_tag_value.get("tag_key"), str)
-                or not isinstance(span_with_tag_value.get("tag_value"), str)
-            ):
-                raise TypeError(
-                    "`span_with_tag_value` must be a dict with keys 'tag_key' and 'tag_value' containing string values"
+        try:
+            has_exactly_one_joining_key = (span is not None) ^ (span_with_tag_value is not None)
+
+            if not has_exactly_one_joining_key:
+                error = "provided_both_span_and_joining_key"
```

Comment:
```suggestion
                error = "provided_both_span_and_tag_joining_key"
```
nit, since span/trace and tag are both joining keys